gstreamerTypes.hpp

```diff
@@ -108,6 +108,131 @@ namespace gstreamer {
          */
         bool binary_in = false;
     };
+
+    /** Response from each connected peer to the RTP pipeline */
+    struct ReceivedResponse {
+        /** The SSRC of this source */
+        int rb_ssrc;
+        /** The SSRC of the sender of this RR */
+        int rb_sender_ssrc;
+        /** Lost 8-bit fraction */
+        int rb_fractionlost;
+        /** Lost packets */
+        int rb_packetslost;
+        /** Highest received seqnum */
+        int rb_exthighestseq;
+        /** Reception jitter (in clock rate units) */
+        int rb_jitter;
+        /** Last SR time (seconds in NTP Short Format, 16.16 fixed point) */
+        int rb_lsr;
+        /** Delay since last SR (seconds in NTP Short Format, 16.16 fixed point) */
+        int rb_dlsr;
+        /** The round-trip time (seconds in NTP Short Format, 16.16 fixed point) */
+        int rb_round_trip;
+    };
+
+    /** The RTP Stream Statistics
+     *
+     * This is the response when the stats is present on the pipeline
+*/
+    struct RTPStreamStatistics {
+        /** The label of the stream */
+        std::string label;
+
+        // The following fields are always present.
+        /** SSRC of this source */
+        int ssrc;
+        /** Confirmation if the source is a source of the session */
+        bool internal;
+        /** Confirmation if source is validated*/
+        bool validated;
+        /** Confirmation if a bye has been received */
+        bool received_bye;
+        /** Confirmation if the source was found as CSRC */
+        bool is_csrc;
+        /** Confirmation if the source is a sender */
+        bool is_sender;
+        /** First seqnum if known */
+        int seqnum_base;
+        /** Clock rate of the media */
+        int clock_rate;
+
+        // The following fields are only present when known.
+        /** Origin of last received RTP Packet - Only present when known */
+        std::string rtp_from;
+        /** Origin of last received RTP Packet - Only present when known */
+        std::string rtcp_from;
+
+        // The following fields make sense for internal sources and will only increase
+        // when "is-sender" is TRUE.
+        /** Number of payload bytes we sent */
+        int octets_sent;
+        /** Number of packets we sent */
+        int packets_sent;
+
+        // The following fields make sense for non-internal sources and will only increase
+        // when "is-sender" is TRUE.
+        /** Total number of payload bytes received */
+        int octets_received;
+        /** Total number of packets received */
+        int packets_received;
+        /** Total number of bytes received including lower level headers overhead */
+        int bytes_received;
```

Comment:
Since there is a difference between senders and receivers ... I think it would be best to have to have two structures that differ for senders and receivers, and have a toplevel stats structures that have an array of each (assuming we can enumerate the streams).

I find it weird that there is only one stream reported by `rtpbin` itself.